coral/src/domain/acl/acl-types.ts

```diff
@@ -0,0 +1,42 @@
+import { KlawApiRequest } from "types/utils";
+
+// Several types are dependent on topictype when it is "Consumer":
+// - aclPatternType can only be "LITERAL"
+// - consumergroup is required
+// - transactionalId becomes available
+// So we define two types depending on the topictype
+
+type CreateAclRequestTopicTypeProducer = Pick<
+  KlawApiRequest<"createAclRequest">,
+  | "remarks"
+  | "aclPatternType"
+  | "topicname"
+  | "environment"
+  | "teamname"
+  | "aclIpPrincipleType"
+  | "consumergroup"
+  | "acl_ssl"
+  | "acl_ip"
+> & { topictype: "Producer" };
+
+type CreateAclRequestTopicTypeConsumer = Pick<
+  KlawApiRequest<"createAclRequest">,
+  | "remarks"
+  | "aclPatternType"
+  | "topicname"
+  | "environment"
+  | "teamname"
+  | "aclIpPrincipleType"
+  | "acl_ssl"
+  | "acl_ip"
+  | "transactionalId"
+> & {
+  topictype: "Consumer";
+  aclPatternType: "LITERAL";
+  consumergroup: string;
+};
+
+export type {
+  CreateAclRequestTopicTypeProducer,
```

Comment:
I assume these will be always used as a union types. Do you think it makes sense to declare the union as its own type and export only type.

Reply:
I was thinking that the individual types could be used for validating the payload, but that validation probably happens at the `zod` level 🤔  Maybe this union type export could be implemented after I have a better view of how it truns out in the end?